wandb/util.py

```diff
@@ -929,9 +929,37 @@ def make_json_if_not_number(
     return json_dumps_safer(v)
 
 
-def make_safe_for_json(obj: Any) -> Any:
+def make_safe_for_json(obj: Any) -> Any:  # noqa: C901
     """Replace invalid json floats with strings. Converts to lists, slices, and dicts.
     Converts numpy array to list. Used for artifact metadata"""
+
+    # Copied from json_friendly logic for handling tensor types
```

Comment:
`json_friendly` has good test coverage; can we either use it directly, or refactor out the common code?

Reply:
Yeah this is actually a really good idea!

Reply:
I can add a parameter flag to json_friendly that handles everything that is required for artifact json, ie.
def json_friendly(obj, artifact_requirements: Optional):
 if artifact_requirements:
   handle len(numpy_arr) > 32
   handle float("-inf") = "-Infinity" etc.